src/dev.rs

```diff
@@ -407,24 +408,27 @@ impl<F: FieldExt> MockProver<F> {
 
         let mut cs = ConstraintSystem::default();
         let config = ConcreteCircuit::configure(&mut cs);
+        let cs = cs;
 
-        let fixed = vec![vec![None; n as usize]; cs.num_fixed_columns];
-        let advice = vec![vec![None; n as usize]; cs.num_advice_columns];
-        let permutations = cs
-            .permutations
-            .iter()
-            .map(|p| permutation::keygen::Assembly::new(n as usize, p))
-            .collect();
+        if n < cs.minimum_rows() {
+            return Err(Error::NotEnoughRowsAvailable);
+        }
+
+        let fixed = vec![vec![None; n]; cs.num_fixed_columns];
+        let advice = vec![vec![None; n]; cs.num_advice_columns];
+        let permutation = permutation::keygen::Assembly::new(n, &cs.permutation);
 
+        let blinding_factors = cs.blinding_factors();
```

Comment:
And then:
```suggestion
        let unusable_rows = cs.minimum_rows() - 1;
```

Reply:
I don't want to use `minimum_rows()` like this. We may need to change `minimum_rows()` later for a reason that has nothing to do with how many rows are usable within ordinary advice columns, for instance.

Reply:
See also [my comment](https://github.com/zcash/halo2/pull/316#discussion_r667347262) about the role of `minimum_rows()`.